### ISSUE TITLE
Fix monitoring BPF maps

### ISSUE THREAD

**Issue by lambdanis** [cilium/tetragon#1774]
Tetragon exposes metrics about BPF maps, defined in the [mapmetrics package](https://github.com/cilium/tetragon/tree/main/pkg/metrics/mapmetrics). There are a few issues with them:
* [ ] Map capacity is exposed as [total](https://github.com/cilium/tetragon/blob/3326a78fbb120a9f5946b64733f8786d08be4b0e/pkg/metrics/mapmetrics/mapmetrics.go#L22) label in `tetragon_map_in_use_gauge` metric (and also, for eventcache it's set to `"0"`). This is against OpenMetrics conventions and makes it difficult (or impossible) to calculate the map utilization. Map capacity should be exposed as a separate metric. Alternatively, Tetragon could calculate the map utilization itself, but exposing the map capacity seems better.
* [ ] The `tetragon_map_in_use_gauge` metric name is not very intuitive IMO and the `gauge` suffix is unnecessary.
* [ ] `processLru` and `eventcache` are not BPF map, they're user-space caches, but info about them is exposed via mapmetrics too. This is confusing, it should be exposed as separate metrics.
* [ ] `tetragon_map_errors_total` metric seems to have an incorrect [help text](https://github.com/cilium/tetragon/blob/3326a78fbb120a9f5946b64733f8786d08be4b0e/pkg/metrics/mapmetrics/mapmetrics.go#L26).
* [ ] `tetragon_map_drops_total` [metric](https://github.com/cilium/tetragon/blob/3326a78fbb120a9f5946b64733f8786d08be4b0e/pkg/metrics/mapmetrics/mapmetrics.go#L13) is incorrect. We get this from [this](https://github.com/cilium/tetragon/blob/3326a78fbb120a9f5946b64733f8786d08be4b0e/pkg/process/cache.go#L134-L136) callback. But this is not called only on evictions, it is also called when we remove elements normally, as we do [here](https://github.com/cilium/tetragon/blob/3326a78fbb120a9f5946b64733f8786d08be4b0e/pkg/process/cache.go#L78). The evictions are counted be the `tetragon_errors_total{type="process_cache_evicted"}` metric (defined [here](https://github.com/cilium/tetragon/blob/82931c612191f391d32375f6d8453666f3cb24d5/pkg/metrics/errormetrics/errormetrics.go#L21)), so `tetragon_map_drops_total` can be just removed.
* [ ] Metrics are not exposed for all maps: https://github.com/cilium/tetragon/issues/1775

**Comment by kkourt** [comment]
@sadath-12 I've unassigned you from this issue because there are many different tasks that can happen in parallel I think. For example, @tpapagian already has a PR to address https://github.com/cilium/tetragon/pull/1949. If you want to work on this, I'd suggest pick a specific bullet point and we can create a separate issue for that.

Thanks!
